rich/live.py

```diff
@@ -0,0 +1,347 @@
+from .control import Control
+from .jupyter import JupyterMixin
+from .live_render import LiveRender
+from .progress import _FileProxy
+from .segment import Segment
+
+
+class _RefreshThread(Thread):
+    """A thread that calls refresh() at regular intervals."""
+
+    def __init__(self, live: "Live", refresh_per_second: float) -> None:
+        self.live = live
+        self.refresh_per_second = refresh_per_second
+        self.done = Event()
+        super().__init__()
+
+    def stop(self) -> None:
+        self.done.set()
+
+    def run(self) -> None:
+        while not self.done.wait(1 / self.refresh_per_second):
+            self.live.refresh()
+
+
+class Live(JupyterMixin, RenderHook):
+    def __init__(
+        self,
+        renderable: RenderableType = "",
+        *,
+        console: Console = None,
+        auto_refresh: bool = True,
+        refresh_per_second: float = 4,
+        transient: bool = False,
+        redirect_stdout: bool = True,
+        redirect_stderr: bool = True,
+        hide_overflow: bool = True,
+    ) -> None:
+        """Renders an auto-updating live display of any given renderable.
+
+        Args:
+            renderable (RenderableType, optional): [The renderable to live display. Defaults to displaying nothing.
+            console (Console, optional): Optional Console instance. Default will an internal Console instance writing to stdout.
+            auto_refresh (bool, optional): Enable auto refresh. If disabled, you will need to call `refresh()` or `update()` with refresh flag. Defaults to True
+            refresh_per_second (float, optional): Number of times per second to refresh the live display. Defaults to 1.
+            transient (bool, optional): Clear the renderable on exit. Defaults to False.
+            redirect_stdout (bool, optional): Enable redirection of stdout, so ``print`` may be used. Defaults to True.
+            redirect_stderr (bool, optional): Enable redirection of stderr. Defaults to True.
+            hide_overflow (bool, optional): Checks that the renderable isn't too large for terminal and auto-hides. Defaults to True.
+        """
+        assert refresh_per_second > 0, "refresh_per_second must be > 0"
+        self.console = console if console is not None else get_console()
+        self._live_render = LiveRender(renderable)
+
+        self._redirect_stdout = redirect_stdout
+        self._redirect_stderr = redirect_stderr
+        self._restore_stdout: Optional[IO[str]] = None
+        self._restore_stderr: Optional[IO[str]] = None
+
+        self._lock = RLock()
+        self.ipy_widget: Optional[Any] = None
+        self.auto_refresh = auto_refresh
+        self._started: bool = False
+        self.transient = transient
+
+        self._refresh_thread: Optional[_RefreshThread] = None
+        self.refresh_per_second = refresh_per_second
+
+        self.hide_overflow = hide_overflow
+        self._hide_render = LiveRender("Terminal too small\n")
```

Comment:
This is problematic. I doubt the dev will want the output to disappear, and it may not even be possible for the user to enlarge the terminal so the output fits. It's also presumptuous to assume the user can read English.

I see the problem though. I think the only thing that we can do at this level is to crop the output, and document the behaviour.

Reply:
I see, so it should be an optional flag to `crop_overflow`? User may also want the whole renderable to be printed no matter what.

Reply:
What actually happens when the output is taller than the terminal? If there is a chance the developer may want that behaviour, then maybe a flag would be appropriate.

Reply:
Actually a `vertical_overflow` flag that supports crop, ellipsis, ignore, would make it similar to the typical `overflow` feature.